node/connection_base.js

```diff
@@ -230,6 +237,13 @@ TChannelConnectionBase.prototype.buildResponse = function buildResponse(req, opt
             state: req.res.state
         }));
     }
+    options = extend({
```

Comment:
move this stuff into `buildOutResponse` instead

Also just pass `inreq` to outResponse as a field if its not already tehre